crates/air_r_formatter/src/comments.rs

```diff
@@ -351,6 +354,184 @@ fn handle_argument_comment(comment: DecoratedComment<RLanguage>) -> CommentPlace
     CommentPlacement::Default(comment)
 }
 
+/// Handle comments close to an argument hole
+///
+/// Hole comment handling is complicated by the fact that holes don't have any
+/// associated tokens. This has two important consequences that we exploit:
+///
+/// - Comments are never enclosed by holes, i.e. `enclosing_node()` never
+///   returns a hole.
+///
+/// - Comments will ALWAYS trail the hole by default, i.e. `preceding_node()`
+///   is how you access the hole node connected to the comment.
+///
+/// Here is the logic for placing hole comments, assuming that we've already
+/// determined that the `preceding_node()` is a hole. Note that this logic is
+/// symmetric, which we quite like:
+///
+/// - If the following token is `,`, `)`, `]`, or `]]`, the comment is
+///   "inside" the hole.
+///
+///   - If the previous sibling node before the hole exists and is not another
+///     hole, attach the comment as trailing trivia on that.
+///
+/// - Else the comment is "after" the hole.
+///
+///   - If the following sibling node after the hole exists and is not another
+///     hole, attach the comment as leading trivia on that.
+///
+/// - Otherwise attach the comment as leading on the hole itself. This happens
+///   when there is not another preceding/following node, or that node is
+///   another hole.
+///
+/// Note that the "following token" check skips following whitespace and
+/// comments, which is what we want.
+///
+/// ## Comments "inside" the hole
+///
+/// Comment trails `a`. Following token is `,` and previous-sibling of hole is a
+/// non-hole.
+///
+/// ```r
+/// fn(
+///   a,<hole> # comment
+///   ,
+///   b
+/// )
+/// ```
```

Comment:
Holes and comments are quite tricky to think about. In the following example:

```r
fn(
  a, # comment
  ,
  b
)
```

you might think the formatter assumes the hole is here

```r
fn(
  a, # comment
  <hole>,
  b
)
```

That would make `# comment` trailing on `a` automatically with the `Default` comment placement behavior.

But that isn't what happens! Instead, holes are placed "eagerly" like this:

```r
fn(
  a,<hole># comment
  ,
  b
)
```

This causes `# comment` to be a trailing comment on the hole by default. This trailing comment behavior ALWAYS happens with holes, and is something we exploit as it simplifies our hole comment handling. It is also how the biome group handles array hole comments, so I feel good about it:
https://github.com/biomejs/biome/blob/66458a12a048b0e686b242d9d4dd0f0e90d1f92a/crates/biome_js_formatter/src/comments.rs#L241-L242